### PR TITLE
applications: nrf_desktop: Bond search & erase triggers

### PR DESCRIPTION
**Pull request by pdunaj**

### CODE REVIEW

File: scripts/hid_configurator/configurator_cli.py [hunk 1]
```diff
@@ -75,18 +75,21 @@ def perform_config(dev, args):
...
+            config_value = None
+        else:
+            try:
+                config_value = value_type(args.value)
+            except ValueError:
+                print('Invalid type for {}. Expected {}'.format(config_name, value_type))
+                return
         success = change_config(dev, recipient, config_name, config_value, device_options, module_id)
 
         if success:
```
Comment (MarekPieta) [inline scripts/hid_configurator/configurator_cli.py]: Maybe update print conditionally:
```
if value_type is None:
    print('{} started') / print('{} set') 
else:
    print('{} set to {}'.format(config_name, config_value))
```

File: applications/nrf_desktop/src/modules/ble_bond.c [hunk 2]
```diff
@@ -750,6 +758,45 @@ static void selector_event_handler(const struct selector_event *event)
...
+	       (MOD_FIELD_GET(config_id) == SETUP_MODULE_BLE_BOND);
+}
+
+static void config_event_handler(const struct config_event *event)
+{
+	if (!is_my_config_id(event->id)) {
+		return;
+	}
+
+	if (state != STATE_IDLE) {
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/ble_bond.c]: The plan was to keep the same restrictions as for button based erase. I don't think we should allow dongle bond erasing using this interface.

File: applications/nrf_desktop/src/modules/ble_bond.c [hunk 3]
```diff
@@ -831,6 +878,17 @@ static bool event_handler(const struct event_header *eh)
 		return false;
 	}
 
+	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE) &&
+	    is_config_event(eh)) {
+		config_event_handler(cast_config_event(eh));
+		return false;
+	}
+
+	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE) &&
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/ble_bond.c]: no - this is leftover

[the remark this answers is not in the document]
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/ble_bond.c]: Fixed

File: applications/nrf_desktop/src/modules/ble_bond.c [hunk 4]
```diff
@@ -750,6 +758,45 @@ static void selector_event_handler(const struct selector_event *event)
...
+		if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
+			erase_adv_confirm();
+			state = STATE_ERASE_ADV;
+		} else if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
+			erase_confirm();
+		}
+		break;
+
+	default:
+		LOG_WRN("Unsupported config event 0x%" PRIx8, event->id);
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/ble_bond.c]: Nothing that comes from outside should be handled as an error. We simply should ignore the wrong parameters. Testing should verify if we use the right values from the script.

File: scripts/hid_configurator/configurator_cli.py (hunk 1 above)
Comment (pdunaj) [reply scripts/hid_configurator/configurator_cli.py]: right and done

**Comment by pdunaj** [comment]
@jhn-nordic , @wiminordic , could you give a green tick?

File: applications/nrf_desktop/src/modules/ble_bond.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_bond.c]: ok

File: applications/nrf_desktop/src/modules/ble_bond.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/ble_bond.c]: ok

**Pull request merged**